packages/shared/hooks/useAsync.ts

```diff
@@ -64,33 +64,37 @@ export function useAsync<R, T extends Function>(cb?: AsyncCb<R, T>) {
     makeEmptyAttempt()
   );
 
-  const run = async (...p: Parameters<AsyncCb<R, T>>) => {
-    try {
-      setState({
-        ...state,
-        status: 'processing',
-      });
-
-      const data = (await cb.call(null, ...p)) as R;
-
-      setState({
-        ...state,
-        status: 'success',
-        data,
-      });
-
-      return [data, null] as [R, Error];
-    } catch (err) {
-      setState({
-        ...state,
-        status: 'error',
-        statusText: err.message,
-        data: null,
-      });
-
-      return [null, err] as [R, Error];
-    }
-  };
+  const run = (...p: Parameters<AsyncCb<R, T>>) =>
```

Comment:
I'm used to always wrapping any promise-returning function within a promise context. Otherwise any error thrown inside `cb` will behave like a regular error, whereas when it's wrapped within the promise context it'll cause the promise to get rejected instead.